src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/AndroidX509CertificateReader.cs

```diff
@@ -0,0 +1,492 @@
+
+        public string Issuer
+        {
+            get
+            {
+                if (_issuer == null)
+                {
+                    // IssuerName is mutable to callers in X509Certificate. We want to be
+                    // able to get the issuer even if IssuerName has been mutated, so we
+                    // don't use it here.
+                    _issuer = Interop.AndroidCrypto.X509GetIssuerName(_cert).Name;
+                }
+
+                return _issuer;
+            }
+        }
+
+        public string Subject
+        {
+            get
+            {
+                if (_subject == null)
+                {
+                    // SubjectName is mutable to callers in X509Certificate. We want to be
+                    // able to get the subject even if SubjectName has been mutated, so we
+                    // don't use it here.
+                    _subject = Interop.AndroidCrypto.X509GetSubjectName(_cert).Name;
+                }
+
+                return _subject;
+            }
+        }
+
+        public string LegacyIssuer => IssuerName.Decode(X500DistinguishedNameFlags.None);
+
+        public string LegacySubject => SubjectName.Decode(X500DistinguishedNameFlags.None);
+
+        public byte[] Thumbprint => Interop.AndroidCrypto.X509GetThumbprint(_cert);
+
+        public string KeyAlgorithm => new Oid(Interop.AndroidCrypto.X509GetPublicKeyAlgorithm(_cert)).Value!;
+
+        public byte[] KeyAlgorithmParameters => Interop.AndroidCrypto.X509GetPublicKeyParameterBytes(_cert);
+
+        public byte[] PublicKeyValue
+        {
+            get
+            {
+                // AndroidCrypto returns the SubjectPublicKeyInfo - extract just the SubjectPublicKey
+                byte[] bytes = Interop.AndroidCrypto.X509GetPublicKeyBytes(_cert);
+                return SubjectPublicKeyInfoAsn.Decode(bytes, AsnEncodingRules.DER).SubjectPublicKey.ToArray();
+            }
+        }
+
+        public byte[] SerialNumber => Interop.AndroidCrypto.X509GetSerialNumber(_cert);
+
+        public string SignatureAlgorithm => Interop.AndroidCrypto.X509GetSignatureAlgorithm(_cert);
+
+        public DateTime NotAfter
+        {
+            get
+            {
+                ulong msFromUnixEpoch = Interop.AndroidCrypto.X509GetNotAfter(_cert);
+                return DateTime.UnixEpoch.AddMilliseconds(msFromUnixEpoch).ToLocalTime();
+            }
+        }
+
+        public DateTime NotBefore
+        {
+            get
+            {
+                ulong msFromUnixEpoch = Interop.AndroidCrypto.X509GetNotBefore(_cert);
+                return DateTime.UnixEpoch.AddMilliseconds(msFromUnixEpoch).ToLocalTime();
+            }
+        }
+
+        public byte[] RawData => Interop.AndroidCrypto.EncodeX509(_cert);
+
+        public int Version
+        {
+            get
+            {
+                int version = Interop.AndroidCrypto.X509GetVersion(_cert);
+                if (version < 0)
```

Comment:
Hm.  Java 9 says this is the encoded version (0, 1, 2), Java 10 says it's the semantic version (1, 2, 3).

What are we using in Android?

https://docs.oracle.com/javase/9/docs/api/javax/security/cert/X509Certificate.html#getVersion--
https://docs.oracle.com/javase/10/docs/api/java/security/cert/X509Certificate.html#getVersion()

Reply:
Android uses the semantic version (1, 2, 3): https://developer.android.com/reference/java/security/cert/X509Certificate#getVersion()

Apparently I verified this property return the expected thing, but missed updating the error check - will fix.